client/controllers/eventMap.coffee

```diff
@@ -59,33 +59,44 @@ Template.eventMap.rendered = ->
 
     if _.isObject query
       allEvents = instance.data.events.find(query, {sort: {creationDate: -1}}).fetch()
-      startingPosition = currentPage * instance.eventsPerPage
       totalEventCount = allEvents.length
+      startingPosition = currentPage * instance.eventsPerPage
     else
       map.removeLayer(markers)
       return
     mapLocations = {}
     templateEvents = []
     eventIndex = startingPosition
 
-    if allEvents.length
+    if totalEventCount
+      filteredEvents = []
       colorScale = chroma.scale(MapHelpers.getDefaultGradientColors()).colors(instance.eventsPerPage)
-
-      while templateEvents.length < instance.eventsPerPage and eventIndex < totalEventCount
-        event = allEvents[eventIndex]
-        eventLocations = instance.data.locations.find({userEventId: event._id}).fetch()
-
-        if eventLocations.length
+      
+      #Remove events that have no locations to plot on the map
```

Comment:
It's way easier to read verbose comments when they have a space after the comment symbol.
I found it to be useful to comment code out without a space and put a space and make the first letter capital when comment contains natural language.